library/debug.c

```diff
@@ -410,4 +410,46 @@ void mbedtls_debug_printf_ecdh( const mbedtls_ssl_context *ssl, int level,
 }
 #endif /* MBEDTLS_ECDH_C */
 
+#if defined(MBEDTLS_SSL_PROTO_TLS1_3) && defined(MBEDTLS_SSL_SESSION_TICKETS)
+#define BITS_OF( var ) ( sizeof( var ) * 8 )
+
+static const char *ticket_flag_name_table[BITS_OF(mbedtls_ssl_tls13_ticket_flags)] =
+{
+    [0] = "ALLOW_PSK_RESUMPTION",
+    [2] = "ALLOW_PSK_EPHEMERAL_RESUMPTION",
+    [3] = "ALLOW_EARLY_DATA",
+};
+
+void mbedtls_debug_print_ticket_flags(
+         const mbedtls_ssl_context *ssl, int level,
+         const char *file, int line,
+         mbedtls_ssl_tls13_ticket_flags flag )
+{
+    char str[DEBUG_BUF_SIZE];
+    size_t i;
+
+    if( NULL == ssl              ||
+        NULL == ssl->conf        ||
+        NULL == ssl->conf->f_dbg ||
+        level > debug_threshold )
+    {
+        return;
+    }
+
+    mbedtls_snprintf( str, sizeof( str ),
+                      "print ticket_flags (0x%02x)\n", flag );
+    debug_send_line( ssl, level, file, line, str );
+
+    for( i = 0; i < BITS_OF(flag); i++ )
+    {
+        if ( flag & ( 1 << i ) )
```

Comment:
```suggestion
        if ( ( flag & ( 1 << i ) ) & MBEDTLS_SSL_TLS1_3_TICKET_FLAG_MASK )
```
there might have typo error. fix it when commit.